srtcore/handshake.h

```diff
@@ -227,23 +227,34 @@ enum UDTRequestType
 
     // Errors reported by the peer, also used as useless error codes
     // in handshake processing functions.
-    URQ_FAILURE_TYPES = 1000
+    URQ_FAILURE_TYPES = 1000,
 
     // NOTE: codes above 1000 are reserved for failure codes for
     // rejection reason, as per `SRT_REJECT_REASON` enum. DO NOT
-    // add any new values here.
+    // add any new values here (below 1000).
+
+    // This is in order to return standard error codes for server
+    // data retrieval failures.
+    URQ_SERVER_FAILURE_TYPES = 2000,
+
+    // This is for a completely user-defined reject reasons.
+    URQ_USER_FAILURE_TYPES = 3000
 };
 
-inline UDTRequestType URQFailure(SRT_REJECT_REASON reason)
+inline UDTRequestType URQFailure(int reason)
 {
     return UDTRequestType(URQ_FAILURE_TYPES + int(reason));
 }
 
-inline SRT_REJECT_REASON RejectReasonForURQ(UDTRequestType req)
+inline int RejectReasonForURQ(UDTRequestType req)
 {
-    if (req < URQ_FAILURE_TYPES || req - URQ_FAILURE_TYPES >= SRT_REJ__SIZE)
+    if (req < URQ_FAILURE_TYPES)
         return SRT_REJ_UNKNOWN;
-    return SRT_REJECT_REASON(req - URQ_FAILURE_TYPES);
+
+    if (req < URQ_SERVER_FAILURE_TYPES && req - URQ_FAILURE_TYPES >= SRT_REJ_E_SIZE)
+        return SRT_REJ_UNKNOWN;
```

Comment:
I think the second if can be simplified to just:
```c++
if (req - URQ_FAILURE_TYPES >= SRT_REJ_E_SIZE)
    return SRT_REJ_UNKNOWN;
```

Reply:
Well, right.

Reply:
Well, wrong. This will change the behavior when `req > URQ_SERVER_FAILURE_TYPES`. This condition is in order to embrace the range between `SRT_REJ_E_SIZE` and `SRT_REJC_SERVER` (for the value of `req - URQ_FAILURE_TYPES`).